### ISSUE TITLE
Static

### ISSUE THREAD

**Issue by jbunce92** [mikebrady/shairport-sync#552]
Hi Mike, 

I love using shairport-sync - works a treat with my hifi. Crystal clear when using iPhone but when using Mac book pro and punchy songs I get random static sounds on the beat. it sounds like a broken cable sort of crackle but obviously thats been ruled out! The same song on an iPhone is fine ... could this be an isolate MBP static issue with the RPI?

Im using USB output to a DAC built into my amp!

thanks for your heroic work!
James

**Comment by mikebrady** [comment]
Thanks for the post. I doubt if it's an MBP – Pi problem. "Random static sounds on the beat" makes me wonder if the DAC is being over-driven. I suggest turning off the Sound Check, Sound Enhancer, Crossfade and the Equaliser in iTunes and listen to a really clean song – if you can find an uncompressed or losslessly compressed one (Apple Lossless), so much the better. The equalizer and the soundcheck options certainly have caused unpleasant audible effects for me, and it drove me crazy until I finally tracked the problem down. Please let us know how you get on...

**Comment by jbunce92** [comment]
Thanks for the reply. Sadly I don't have any of those features on, but oddly do on my phone which doesn't suffer the same issue - perhaps ill try with them on! The only other thing I had though was could there be some interference with streaming music - RPI is attached to wifi hub via an ethernet. Most of my phone music is downloaded, where as my MBP music is streamed... my technical knowledge is pretty much exhausted at this stage! 
Clean, gentle, music is always ok, its albums like Fleetwood Mac - Rumours, that cause the problem, and even then its not consistent.

**Comment by mikebrady** [comment]
Fair enough, thanks.

The thing is, I t is possible that the streamed version of a track is compressed differently to your own, downloaded, version.

To make the comparison between the iPhone and MBP valid, you should put exactly the same track in your MBP and iPhone and do the listening tests then.

I have never been able to detect an audible difference between the Music app on the iPhone and iTunes on a Mac or PC on [exactly] identical source material, so long as there are no transmission errors and all those settings mentioned earlier are disabled.

**Comment by rdain** [comment]
Thanks for opening up this topic. And thanks for your software and docs on it Mike, it's amazing how well it works.

But I had something similar (?) happen just this week. I didn't think to test by replaying the track itself; the sound is more like a hard buzzing crackle that honestly could be bad for the speakers. Wasn't sure what the hell was going on, but the details: I'm running on Raspberian Jessie (`Linux kitchen 4.9.28-v7+ #998 SMP`), output on a HiFiBerry DAC+, and though I hadn't any serious trouble before, while I was home playing from iTunes on laptop (based on iTunes' own tracks from the album Converting Vegetarians) I'd get a sudden high-gain buzzing/crackle sound. It would do one of two things: come in and go out suddenly; or repeat a few times (no discernible pattern, just random buzzing crackling sounds) and then lose connectivity; if the sound fell out, it would pick back up again after a few seconds.

Some of the sound reminded me a bit of the same sort of sound I'd hear from time to time when a cell phone would receive a call near a desk speaker, and buzzing would come in over the speaker. But there's nothing near the analog output line that would generate that sort of activity, or explain how loud it was. Not sure what was causing the issue. The device is currently running data via ethernet over power. Reading this, I would've thought it was about the iTunes equalizer as well, but it happened the next day as well from my iPhone 6. So ... yeah, I've only got a bunch of red herrings here.

Is there a good way to monitor the signal running across the daemon? I'd be curious to set up a munin collector or maybe even a Nagios plugin to keep an eye on it.

**Comment by mikebrady** [comment]
Thanks for the post. The key to identifying this problem – whatever the cause of it, whether it's Shairport Sync or something else – is being able to generate the issue. If you could reliably make the problem happen, then we could begin narrow down the potential causes.

**Comment by jbunce92** [comment]
Just to close the initial issue, I think I've found the solution. Essentially Mike you were right - clipping was to blame! Completely resolved when I lowered the volume on the MBP/Ipad and I guess they just have a higher max output via airplay than an iPhone which is why it wasn't an issue on my iPhone. To make the solution permanent I just turned down the alsamixer volume to 70% - voila! Queerbsd I don't know if you've tried such a simple solution...?
